opal/mca/pmix/pmix3x/pmix3x.c

```diff
@@ -1189,9 +1189,11 @@ int pmix3x_value_unload(opal_value_t *kv,
             /* handle the various types */
             if (PMIX_INFO == v->data.darray->type) {
                 pmix_info_t *iptr = (pmix_info_t*)v->data.darray->array;
-                if (NULL != iptr[n].key) {
-                    ival->key = strdup(iptr[n].key);
+                if (NULL == iptr) {
+                    rc = OPAL_ERR_BAD_PARAM;
+                    break;
                 }
+                ival->key = strdup(iptr[n].key);
```

Comment:
Do you still want to protect strdup() with a check of (iptr[n].key != NULL) also?  I think strdup(NULL) can segv on some systems.

Reply:
No - iptr[n].key is a statically defined array and so it cannot be NULL